dashboard/src/App.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import DeploymentOperator from './deployment/DeploymentOperator.js';
+import NoOperator from './NoOperator.js';
+import { apiGet } from './api/api.js';
+import { Container, Segment, Message } from 'semantic-ui-react';
+import './App.css';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {operators:{}};
+  }
+
+  async componentDidMount() {
+    this.reloadOperators();
+  }
+
+  async reloadOperators() {
+    const operators = await apiGet('/api/operators');
+    this.setState({operators});
+  }
+
+  render() {
+    setTimeout(this.reloadOperators.bind(this), 5000);
```

Comment:
This should happen in componentDidMount. Having side-effects in a render function is generally a bad idea.

Reply:
Also I think you want to use setInterval as setTimeout is for one-off delayed code.

Reply:
Also keep in mind setTimeout/setInterval will still keep firing if the component is unmounted. So you need to clean those up in componentWillUnmount.